apis/pipeline/pipeline.yaml

```diff
@@ -4516,6 +4535,37 @@ components:
                     description: Node pool update process ID.
                     type: string
 
+        ListNodePoolsResponse:
```

Comment:
This should probably be some generic NodePool object (that could be used for a potential GET as well). This is related for the comment above.

Reply:
Fixed in https://github.com/banzaicloud/pipeline/compare/2ad4cafb067c442a1cfbf1b9cd4071e298b269c8..f640984ef5dee7ae5770cc55c9ccc9741c0e760b.

There was already a [`NodePool`](https://github.com/banzaicloud/pipeline/blob/f640984ef5dee7ae5770cc55c9ccc9741c0e760b/apis/pipeline/pipeline.yaml#L4452) object defined which is a `oneOf` [`EksNodePool`](https://github.com/banzaicloud/pipeline/blob/f640984ef5dee7ae5770cc55c9ccc9741c0e760b/apis/pipeline/pipeline.yaml#L4497) and possibly other provider specific node pool objects in the future which can share a generic [`GenericNodePool`](https://github.com/banzaicloud/pipeline/blob/f640984ef5dee7ae5770cc55c9ccc9741c0e760b/apis/pipeline/pipeline.yaml#L4456) base holding name, size and labels.

This was already present in the pre-feature code so I used those.

Missed the opportunity while I took over the feature code as I supposed the existing feature code implementation would have been based on such foundations.
Thanks for the note, it feels much better this way.